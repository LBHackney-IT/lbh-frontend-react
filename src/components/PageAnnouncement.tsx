import classNames from "classnames";
import PropTypes from "prop-types";
import React from "react";

import "lbh-frontend/lbh/components/lbh-page-announcement/_page-announcement.scss";

/**
 * The proptypes for {@link PageAnnouncement}.
 */
export interface PageAnnouncementProps {
  id?: string;
  className?: string;
  /**
   * The title of the announcement
   */
  title: string;
  /**
   * The content of the announcement
   */
  children: React.ReactNode;
}

/**
 * A component for displaying announcements on a page.
 */
export const PageAnnouncement: React.FunctionComponent<PageAnnouncementProps> = (
  props: PageAnnouncementProps
): JSX.Element => {
  const { id, className, title, children } = props;

  return (
    <section id={id} className={classNames("lbh-page-announcement", className)}>
      <h3 className="lbh-page-announcement__title">{title}</h3>
      <div className="lbh-page-announcement__content">{children}</div>
    </section>
  );
};

PageAnnouncement.propTypes = {
  id: PropTypes.string,
  className: PropTypes.string,
  title: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired
};
