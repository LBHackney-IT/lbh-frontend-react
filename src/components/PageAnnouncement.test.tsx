import React from "react";
import { create } from "react-test-renderer";

import { PageAnnouncement } from "./PageAnnouncement";

it("renders correctly with all props", () => {
  const component = create(
    <PageAnnouncement id="test" className="class1 class2" title="testTitle">
      children
    </PageAnnouncement>
  );

  expect(component).toMatchSnapshot();
});

it("renders correctly without optional props", () => {
  const component = create(
    <PageAnnouncement title="testTitle">children</PageAnnouncement>
  );

  expect(component).toMatchSnapshot();
});
