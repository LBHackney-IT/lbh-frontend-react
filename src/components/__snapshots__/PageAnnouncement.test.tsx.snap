// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with all props 1`] = `
<section
  className="lbh-page-announcement class1 class2"
  id="test"
>
  <h3
    className="lbh-page-announcement__title"
  >
    testTitle
  </h3>
  <div
    className="lbh-page-announcement__content"
  >
    children
  </div>
</section>
`;

exports[`renders correctly without optional props 1`] = `
<section
  className="lbh-page-announcement"
>
  <h3
    className="lbh-page-announcement__title"
  >
    testTitle
  </h3>
  <div
    className="lbh-page-announcement__content"
  >
    children
  </div>
</section>
`;
